tests/unit/backend/dataio/test_data.py

```diff
@@ -181,6 +183,31 @@ def test_extract_rotation_angles(data_fixture):
     rst = _extract_rotation_angles([metadata_tiff] * 3)
     ref = np.array([0.1, 0.1, 0.1])
     np.testing.assert_array_almost_equal(rst, ref)
+    # case_2: mixed file types
+    rst = _extract_rotation_angles([good_tiff, metadata_tiff, generic_tiff, generic_fits])
+    ref = np.array([10.02, 0.1, np.nan, np.nan])
+    np.testing.assert_array_equal(rst, ref)
+    # case_3: all files without extractable angles
+    rst = _extract_rotation_angles([generic_tiff, generic_fits])
+    assert rst is None
+
+
+def test_extract_rotation_angle_from_filename():
```

Comment:
Please add some negative tests

Reply:
This `assert extract_rotation_angle_from_filename("generic_file.tiff") is None` is the negative test.